Gym_en.rst

```diff
@@ -0,0 +1,213 @@
+
+- \ ``Accelerate (Acceleration value)`` \: Indicates that the agent is accelerated by the size of \ ``acceleration value`` \. The value range of \ ``Acceleration value`` \ is \ ``[0,1]`` \. The numeric type is \``float32``.
+  
+- \ ``Turn (Rotation value)`` \: means to turn the agent in the direction of \ ``rotation value`` \. The value range of \ ``Rotation value`` \ is \ ``[-1,1]`` \. The numeric type is \``float32``.
+  
+- \ ``Break ()`` \: means stop.
+
+Using the gym environment space definition can be expressed as:
+
+.. code::python
+    
+    from gym import spaces
+
+    action_space = spaces.Tuple((spaces.Discrete(3),
+                                    spaces.Box(low=0, high=1, shape=(1,)),
+                                    spaces.Box(low=-1, high=1, shape=(1,))))
+
+state space
+------------
+
+The state space of Gym-hybrid is represented by a list of 10 elements, which describes the current state of the agent, including the agent's current coordinates, velocity, the sine and cosine of the orientation angle, the coordinates of the target, the distance between the agent and the target, and the target. Distance-related bool value, current relative steps.
+
+.. code::python
+
+    state = [
+                agent.x,
+                agent.y,
+                agent.speed,
+                np.cos(agent.theta),
+                np.sin(agent.theta),
+                target.x,
+                target.y,
+                distance,
+                0 if distance > target_radius else 1,
+                current_step / max_step
+            ]
+
+Bonus space
+-------------
+The reward of each step is set as the length of the agent's distance from the target after the action is performed in the previous step minus the length of the distance from the target after the current step performs the action, ie \``dist_t-1 - dist_t`` \. The algorithm has a built-in \``penalty``\ to motivate the agent to be faster
+Achieve goals. When the episode ends, if the agent stops in the target area, it will get an additional reward with a value of 1; if the agent goes out of bounds or exceeds the maximum number of steps of the episode, it will not get additional reward. The reward at the current moment is expressed by the formula as follows:
+
+.. code::python
+
+    reward = last_distance - distance - penalty + (1 if goal else 0)
+
+
+Termination condition
+------------
+The termination condition for each episode of the Gym-hybrid environment is any of the following:
+
+- The agent successfully entered the target area
+  
+- agant out of bounds
+  
+- Reach the maximum step of the episode
+  
+
+built-in environment
+------------
+There are two built-in environments, \``"Moving-v0"``\ and \``"Sliding-v0"``\. The former does not consider conservation of inertia, while the latter does (so is more practical). The two environments are consistent in state space, action space, and reward space.
```

Comment:
There are two built-in environments,\ ``"Moving-v0"`` \and \ ``"Sliding-v0"`` \. The former does not consider conservation of inertia, while the latter does (so is more practical). The two environments are consistent in state space, action space, and reward space.